reconcile/utils/oc.py

```diff
@@ -620,6 +635,245 @@ def _run_json(self, cmd, allow_not_found=False):
+                        # permisions, so go to the next one
+                        next
+                    # resources are in a kind:ResourceList map, where a
+                    # ResourceList contains the api endpoint
+                    # (apigroup/version/puralkind)
+                    for kind in r.keys():
+                        if len(apigroup) == 0:
+                            gv = f'{version}'
+                        else:
+                            gv = f'{apigroup}/{version}'
+                        kind_groupversion = self.add_group_kind(
+                            kind, kind_groupversion, gv, obj.preferred)
+        return kind_groupversion
+
+    def get_items(self, kind, **kwargs):
+        if not self.use_native:
+            return super().get_items(kind, **kwargs)
+
+        k, group_version = self._parse_kind(kind)
+        obj_client = self._get_obj_client(group_version=group_version, kind=k)
+
+        namespace = ''
+        if 'namespace' in kwargs:
+            namespace = kwargs['namespace']
+            # for cluster scoped integrations
+            # currently only openshift-clusterrolebindings
+            if namespace != 'cluster':
+                if not self.project_exists(namespace):
+                    return []
+
+        labels = ''
+        if 'labels' in kwargs:
+            labels_list = [
+                "{}={}".format(k, v)
+                for k, v in kwargs.get('labels').items()
+            ]
+
+            labels = ','.join(labels_list)
+
+        resource_names = kwargs.get('resource_names')
+        if resource_names:
+            items = []
+            for resource_name in resource_names:
+                try:
+                    item = obj_client.get(
+                        name=resource_name, namespace=namespace,
+                        label_selector=labels)
+                    if item:
+                        items.append(item.to_dict())
+                except NotFoundError:
+                    pass
+            items_list = {'items': items}
+        else:
+            items_list = obj_client.get(
+                namespace=namespace, label_selector=labels).to_dict()
+
+        items = items_list.get('items')
+        if items is None:
+            raise Exception("Expecting items")
+
+        return items
+
+    def get(self, namespace, kind, name=None, allow_not_found=False):
+        if not self.use_native:
+            return super().get(namespace, kind, name, allow_not_found)
+
+        k, group_version = self._parse_kind(kind)
+        obj_client = self._get_obj_client(group_version=group_version, kind=k)
+        try:
+            obj = obj_client.get(name=name, namespace=namespace)
+            return obj.to_dict()
+        except NotFoundError as e:
+            if allow_not_found:
+                return {}
+            else:
+                raise StatusCodeError(f"[{self.server}]: {e}")
```

Comment:
Does `e` mention `name` and `namespace`? Unless they come from a vault, they'd be very useful to have when troubleshooting.

Reply:
+1, very good point. This goes in #1400 though.

Reply:
ATM, this is the same error message raised by the original get function.  I would prefer not to change that here and do it in a separate PR once this is merged.  My intention was to keep inputs/outpus the same.